pkg/kv/kvserver/rangefeed/scheduled_processor.go

```diff
@@ -322,10 +322,7 @@ func (p *ScheduledProcessor) Register(
 		// Construct the catchUpIter before notifying the registration that it
 		// has been registered. Note that if the catchUpScan is never run, then
 		// the iterator constructed here will be closed in disconnect.
-		if err := r.maybeConstructCatchUpIter(); err != nil {
-			r.disconnect(kvpb.NewError(err))
-			return nil
-		}
+		r.maybeConstructCatchUpIter()
```

Comment:
This comment and method name is inaccurate: the iterator has already been constructed by the caller at this point. The same goes for a lot of the other related infrastructure which now appears misnamed.